src/audio/component.c

```diff
@@ -375,6 +375,9 @@ int comp_set_model(struct comp_dev *dev, struct comp_model_data *model,
 	 * memory for whole model data
 	 */
 	if (!cdata->msg_index) {
+		if (!cdata->data->size)
```

Comment:
Why is this valid ? Can we have an inline comment explaining why size 0 models are allowed (otherwise it can look like an erroneous check)

Reply:
@lgirdwood 
I saw that case in #3068.
@keyonjie 
Do you know why requested data size in #3068 is equal to 0?

Reply:
If 0 is invalid size that should be responded with 0, the current path is fine (comp_alloc_model_data() but one enhancement would be useful, `if (!model->data)` error message should print the requested size to know immediately what was the reason.

Reply:
It also seems that comp_alloc_model_data() works as "realloc" (perhaps comp_realloc_model_data() would be more appropriate name). So "free" of existing model if any should be done first, then size == 0 test, so that if "null model" is provided, the previous one is freed.

Reply:
@mmaka1 
I've updated PR. Now ```comp_free_model_data()``` is before size check and there is no ```comp_free_model_data()``` in ```comp_alloc_model_data()```